psyneulink/core/compositions/composition.py

```diff
@@ -9037,16 +9204,19 @@ def run(
                         )
                     node.initialize(initialize_cycle_values[node], context)
 
-        if reset_stateful_functions_to is None:
-            reset_stateful_functions_to = {}
-
-        for node in reset_stateful_functions_to:
-            # make sure the args to reset are in the form of an iterable so they can be unpacked
-            try:
-                iter(reset_stateful_functions_to[node])
-            except TypeError:
-                reset_stateful_functions_to[node] = [reset_stateful_functions_to[node]]
-            node.reset(*reset_stateful_functions_to[node], context=context)
+        try:
+            for node in reset_stateful_functions_to:
+                # make sure the args to reset are in the form of an iterable so they can be unpacked
+                try:
+                    iter(reset_stateful_functions_to[node])
+                except TypeError:
+                    reset_stateful_functions_to[node] = [reset_stateful_functions_to[node]]
```

Comment:
the conversion can be put in the if-block below to avoid unnecessary conversion if it's not used

Reply:
In this case, the conversion shouldn't be in the if-block, because the converted values will be used later, regardless of whether or not conditions are met to trigger the reset here.